### PR TITLE
Fix device-tree assigned-clocks properties parsing

### PR DESCRIPTION
**Pull request by clementleger**
This small series fixes parsing problems when using `assigned-clocks` properties:
- Wrong check for assigned-clock-rates property
- Handling of phandle == 0

### CODE REVIEW

File: core/kernel/dt_driver.c [hunk 1]
```diff
@@ -266,6 +266,11 @@ void *dt_driver_device_from_node_idx_prop(const char *prop_name,
 	while (idx < len) {
 		idx32 = idx / sizeof(uint32_t);
 		phandle = fdt32_to_cpu(prop[idx32]);
+		if (phandle == 0) {
```
Comment (jenswi-linaro) [inline core/kernel/dt_driver.c:269]: While this style isn't rejected, please don't mix it with the other style in the same function.
So make this `if (!phandle)...`

**Review by etienne-lms — COMMENTED** [review]
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>`

File: core/kernel/dt_driver.c (hunk 1 above)
Comment (clementleger) [reply core/kernel/dt_driver.c]: Indeed, I'll fixed that.

**Comment by clementleger** [comment]
New version:
- Use `(!phandle)`.
- Add early return, it worked previously due to some wrong check... (prop_idx being an unsigned, - 1, is not a value we want to handle...)

@etienne-lms I did not added your `Reviewed-by` on the first commit since I modified it.

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by etienne-lms** [comment]
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>` for the 2 commits.

**Pull request merged**